### PR TITLE
Upgrade `automerge` to `0.5`

### PR DESCRIPTION
**Pull request by MarijnS95**

### CODE REVIEW

File: src/de/map.rs [hunk 1]
```diff
@@ -35,7 +38,13 @@ impl<'de, 'a> de::MapAccess<'de> for MapDeserializer<'a> {
     where
         K: de::DeserializeSeed<'de>,
     {
-        if let Some((key, value, id)) = self.values.next() {
+        if let Some(MapRangeItem {
+            key,
+            value,
+            id,
+            conflict: _,
```
Comment (Athosvk) [inline src/de/map.rs:45]: Nit: 
```suggestion
            ..,
```

File: tests/updates.rs [hunk 2]
```diff
@@ -46,6 +47,89 @@ fn test_updates_and_merging() {
...
+        &numbers_id,
+    );
+    println!();
+
+    // Receiving
+    doc_receive.load_incremental(&data).unwrap();
+    let (player_receive2, camera_receive2, numbers_receive2) = receive(&doc_receive);
+
+    // Assert that local and remote changes were merged successfully
+    // XXX: Inline the functions, the code below implies knowing thwat update1() and update2() touched...
```
Comment (Athosvk) [inline tests/updates.rs]:
```suggestion
    // XXX: Inline the functions, the code below implies knowing that update1() and update2() touched...
```
XXX -> TODO? Or not even relevant anymore?

File: tests/updates.rs [hunk 3]
```diff
@@ -46,6 +47,89 @@ fn test_updates_and_merging() {
...
+    update2(
+        &mut doc_receive,
+        &mut player_receive,
+        &mut camera_receive,
+        &mut numbers_receive,
+        &player_id,
+        &camera_id,
+        &numbers_id,
+    );
+    println!();
```
Comment (Athosvk) [inline tests/updates.rs]: Debug code?

File: tests/updates.rs [hunk 4]
```diff
@@ -46,6 +47,89 @@ fn test_updates_and_merging() {
...
+        &mut player_send,
+        &mut camera_send,
+        &mut numbers_send,
+        &player_id,
+        &camera_id,
+        &numbers_id,
+    );
+    let (commit, _patch_log) =
+        tx.commit_with(CommitOptions::default().with_message("Updating position"));
+    let _commit_after_update1 = commit.unwrap();
```
Comment (Athosvk) [inline tests/updates.rs]: Can this be removed?

File: src/de/map.rs (hunk 1 above)
Comment (MarijnS95) [reply src/de/map.rs:45]: I wanted to explicitly make sure all fields are listed, if you don't mind?

File: tests/updates.rs (hunk 2 above)
Comment (MarijnS95) [reply tests/updates.rs]: Nice typo :D

File: src/de/map.rs (hunk 1 above)
Comment (Athosvk) [reply src/de/map.rs:45]: Should have clarified but of course don't mind, I just thought we generally prefer the { ..}  notation

File: tests/updates.rs (hunk 3 above)
Comment (MarijnS95) [reply tests/updates.rs]: I think this is a copy-paste from the original, to separate some data that is printed between `update2()` and `receive()`. But there's no empty line between `update1()` anyway (and I didn't bother removing any logging when converting this to a test, to ease debugging).

File: tests/updates.rs (hunk 4 above)
Comment (MarijnS95) [reply tests/updates.rs]: Maybe, I wanted to show that there is a commit hash here that could be used.

Comment (Athosvk) [reply tests/updates.rs]: Purely talking about the unwrap line. I understand wanting to clarify there's something returned by the `commit_with` function. Should actually be marked as nit anyway, it's nothing blocking

**Review by Athosvk — APPROVED** [review]

**Pull request merged**